syntax = "proto3";
package PlatPmd;
// option optimize_for = LITE_RUNTIME;

// 账号模块协议定义
message AccPmd
{
    enum Param
    {
        LoginAccPmd_C2S    = 0;
        LoginAccPmd_S2C    = 1;
    }
}

enum Error
{
    LoginOk             = 0;    // 登录成功
    TokenFindError      = 1;    // 服务器没有token
    TokenDiffError      = 2;    // token错误
    VersionError        = 3;    // 版本验证
}

// 登录
message LoginAccPmd_C2S
{
    string platform = 1;                // 平台名字
    string game_account_id = 2;         // 游戏的账号Id
    string game_account_sign = 3;       // 游戏的账号Id的签名
    string channel_open_id = 4;         // 渠道的用户Id
    string channel_param = 5;           // 渠道其他参数
}

// 登录返回
message LoginAccPmd_S2C
{
    int32 code = 1;                 // 错误码
    int32 id = 2;                   // 角色id
    string channel_open_id = 3;     // 运营平台的用户id
    string game_account_id = 4;     // 游戏的账号Id
    string game_account_sign = 5;   // 游戏的账号Id的签名
    string game_login_key = 6;      // 登录key 用于重新登录时发送
}
